mineral-ui/mineral-ui
# [Site] update to Netlify hosting

### Description
This PR moves our hosting from GitHub Pages over to Netlify ✨.

connects #203 

The steps after this PR is merged are
1. Go into the netlify console and plug in our deploy scripts.
2. Make sure the builds work
3. Move DNS over

### Motivation and context
Some of the advantages over GH Pages:
- Deploy Previews (this is the big one. free staging site without a different environment config)
- rollbacks
- free ssl
- custom rewrites and redirects

### How to test

1. `npm run clean && npm install`
2. `npm run deploy:site`. The first time you run this, you'll have to authenticate your instance of `netlify-cli` on their site with the `mineral-ci` user.
3. once the deploy finishes, you'll get a link to a netlify site. The site should look the same
4. 404s redirect to `/`, which is done with a webpack-dev-server setting + a netlify config file depending on environment

### Types of changes
- refactor

### Checklist


### How does this PR make you feel?
![Alt text here](https://media.giphy.com/media/qZ8xqTofmU2UE/giphy.gif)
